sdk/python/tests/integration/offline_store/test_feature_logging.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+import uuid
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pytest
+from google.api_core.exceptions import NotFound
+
+from feast.feature_logging import (
+    LOG_DATE_FIELD,
+    LOG_TIMESTAMP_FIELD,
+    REQUEST_ID_FIELD,
+    FeatureServiceLoggingSource,
+    LoggingConfig,
+)
+from feast.feature_service import FeatureService
+from feast.protos.feast.serving.ServingService_pb2 import FieldStatus
+from feast.wait import wait_retry_backoff
+from tests.integration.feature_repos.repo_configuration import (
+    UniversalDatasets,
+    construct_universal_feature_views,
+)
+from tests.integration.feature_repos.universal.entities import driver
+from tests.integration.feature_repos.universal.feature_views import conv_rate_plus_100
+
+
+@pytest.mark.integration
+@pytest.mark.universal
+def test_feature_service_logging(environment, universal_data_sources):
+    store = environment.feature_store
+
+    (_, datasets, data_sources) = universal_data_sources
+
+    feature_views = construct_universal_feature_views(data_sources)
+    store.apply([driver(), *feature_views.values()])
+
+    logs_df = prepare_logs(datasets)
+
+    feature_service = FeatureService(
+        name="test_service",
+        features=[
+            feature_views.driver[["conv_rate", "avg_daily_trips"]],
+            feature_views.driver_odfv[
+                ["conv_rate_plus_val_to_add", "conv_rate_plus_100_rounded"]
+            ],
+        ],
+        logging_config=LoggingConfig(
+            destination=environment.data_source_creator.create_logged_features_destination()
+        ),
+    )
+
+    num_rows = logs_df.shape[0]
+    first_batch = logs_df.iloc[: num_rows // 2, :]
+    second_batch = logs_df.iloc[num_rows // 2 :, :]
+
+    schema = FeatureServiceLoggingSource(
+        feature_service=feature_service, project=store.project
+    ).get_schema(store._registry)
+
+    store.write_logged_features(
+        source=feature_service, logs=pa.Table.from_pandas(first_batch, schema=schema),
+    )
+
+    store.write_logged_features(
+        source=feature_service, logs=pa.Table.from_pandas(second_batch, schema=schema),
+    )
```

Comment:
I didn't see this call happening anywhere else in the code - is my understanding correct that this would need to be added to `get_online_features` before we would log anything?

Reply:
Integration would be added in following PRs. Yes, this function will be called from `ge_online_features`.